### PR TITLE
[FEATURE] added filtering by protein accession stringlist

### PR DESCRIPTION
**Pull request by timosachsenberg**
addresses issue https://github.com/OpenMS/OpenMS/issues/1268

### CODE REVIEW

**Comment by timosachsenberg** [comment]
https://github.com/OpenMS/OpenMS/issues/1320

File: src/openms/source/FILTERING/ID/IDFilter.cpp [hunk 1]
```diff
@@ -351,6 +351,50 @@ namespace OpenMS
...
+
+    for (Size i = 0; i < identification.getHits().size(); i++)
+    {
+      std::set<String> protein_accessions = identification.getHits()[i].extractProteinAccessions();
+      for (set<String>::const_iterator ac_it = protein_accessions.begin(); ac_it != protein_accessions.end(); ++ac_it)
+      {
+        if (std::find(proteins.begin(), proteins.end(), *ac_it) != proteins.end())
+        {
+          filtered_peptide_hits.push_back(identification.getHits()[i]);
+        }
```
Comment (timosachsenberg) [inline src/openms/source/FILTERING/ID/IDFilter.cpp]: forgot break... will update

**Comment by lars20070** [comment]
+1

File: src/topp/IDFilter.cpp [hunk 2]
```diff
@@ -196,6 +196,8 @@ class TOPPIDFilter :
     setValidFormats_("whitelist:proteins", ListUtils::create<String>("fasta"));
     registerFlag_("whitelist:by_seq_only", "Match peptides with FASTA file by sequence instead of accession and disable protein filtering.");
 
+    registerStringList_("whitelist:protein_accessions", "<accessions>", ListUtils::create<String>(""), "All peptides that are not a substring of a sequence in this file are removed\nAll proteins whose accession is not present in this file are removed.", false);
```
Comment (cbielow) [inline src/topp/IDFilter.cpp]: doc incorrect (copy&pasted?)

**Comment by cbielow** [comment]
+1 (when doc is fixed :))

**Comment by timosachsenberg** [comment]
ok I fixed the parameter description and rebased

**Pull request merged**